var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -42,41 +41,64 @@ def url_for_version(self, version):
     variant('fortran', default=True, description='Build the Fortran Interface')
     variant('pic', default=True,
             description='Produce position-independent code (for shared libs)')
+    variant('shared', default=True, description='Enable shared library')
 
     depends_on('mpi')
 
     depends_on('m4', type='build')
 
-    # See:
-    # https://trac.mcs.anl.gov/projects/parallel-netcdf/browser/trunk/INSTALL
-    def configure_args(self):
-        spec = self.spec
+    conflicts('+shared', when='@:1.9%nag+fortran')
+    conflicts('+shared', when='@:1.8')
+
+    patch('nag_libtool.patch', when='@1.10:999%nag')
```

Comment:
I take it this has been fixed upstream? In that case, you might want to do something like:
```suggestion
    patch('nag_libtool.patch', when='@1.10:1.12.1%nag')
```

Reply:
This has not been fixed upstream and hardly will be in the nearest future because it would require a special `autogen` script, which would apply patches to the libtool-generated files. Developers are usually very reluctant to switch from a simple `autoreconf -fvi` to something more complex with an `autogen` script and a bunch of patches for libtool. The reason I introduced the `999` constraint was not to apply the patch to the `develop` version, which does not have files that are modified with the patch.

Reply:
It turns out that I was wrong: there is already a patched version of `libtool.m4` upstream so no `autogen.sh` is really needed. The patch for `ltmain.sh` can be handled the same way: https://github.com/Parallel-NetCDF/PnetCDF/pull/59